router/xgress_edge_transport/dialer.go

```diff
@@ -52,13 +54,13 @@ func (txd *dialer) Dial(destination string, circuitId *identity.TokenId, address
 
 	txDestination, err := transport.ParseAddress(destination)
 	if err != nil {
-		return nil, fmt.Errorf("cannot dial on invalid address [%s] (%s)", destination, err)
+		return nil, xgress.InvalidTerminatorError{InnerError: fmt.Errorf("cannot dial on invalid address [%s] (%s)", destination, err)}
 	}
 
 	log.Debug("dialing")
 	peer, err := txDestination.Dial("x/"+circuitId.Token, circuitId, txd.options.ConnectTimeout, nil)
 	if err != nil {
```

Comment:
Should be a regular dial error

Reply:
This is how I am checking back in the controller if its an "expected" error. The dial error gets persisted in the inner error and checked on the controller side. Does that need to be redone?